### PR TITLE
[PT] Translation for Strings

### PR DESCRIPTION
**Pull request by ruan-brandao**
Related to [#355](https://github.com/doomspork/elixir-school/issues/355)

This lesson is not in the issue's list of pages to translate, but it doesn't have a translation.

Feel free to give me your feedback @pragmaticivan and anyone who wants to revise this translation.

### CODE REVIEW

File: pt/lessons/basics/strings.md [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+lang: pt
+---
+
+Strings, listas de caracteres, Graphemes e Codepoints.
+
+{% include toc.html %}
+
+## Strings
+
+Strings no Elixir são nada mais que uma sequência de bytes. Vamos ver um exemplo:
```
Comment (anderkonzen) [inline pt/lessons/basics/strings.md]: Hi @ruan-brandao! I took the liberty to review and add some notes...

Maybe "Strings **em** Elixir..."?

File: pt/lessons/basics/strings.md [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+"hello"
+```
+
+>NOTA: Ao usar a sintaxe com << >> estamos dizendo ao compilador que os elementos dentro desses símbolos são bytes.
+
+## Listas de Caracteres
+
+Internamente, as strings em Elixir são representadas como uma sequência de bytes ao invés de um array de caracteres. Elixir também tem um tipo char list (lista de caracteres). Strings em Elixir são delimitadas por aspas duplas, enquanto listas de caracteres são delimitadas por aspas simples.
+
+Qual a diferença? Cada valor de uma lista de caracteres é um valor ASCII do caractere. Vamos ver isso mais a fundo:
```
Comment (anderkonzen) [inline pt/lessons/basics/strings.md]: "é **o** valor ASCII"?

File: pt/lessons/basics/strings.md [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+### Anagramas
+
+A e B são considerados anagramas se há alguma forma de rearranjar A ou B para torná-las iguais. Por exemplo:
+
++ A = super
++ B = perus
+
+Se nós rearranjarmos os caracteres da string A, podemos obter a string B, e vice-versa.
+
+Então, como podemos checar se duas strings são Anagramas em Elixir? A solução mais fácil é apenas ordenar os graphemes de cada string em ordem alfabética e então checar se as duas listas são iguais. Vamos tentar:
```
Comment (anderkonzen) [inline pt/lessons/basics/strings.md]: I believe "verificar" is better than "checar"...

File: pt/lessons/basics/strings.md [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+
+iex> Anagram.anagrams?("María", "íMara")
+true
+
+iex> Anagram.anagrams?(3, 5)
+** (FunctionClauseError) no function clause matching in Anagram.anagrams?/2
+    iex:2: Anagram.anagrams?(3, 5)
+```
+
+Como você pode ver, a última chamada a `anagrams?` causou um FunctionClauseError. Esse erro nos diz que não há nenhuma função no nosso módulo de acordo com o padrão de receber dois argumentos não binários, e isso é exatamente o que nós queremos, receber apenas duas strings e nada mais.
```
Comment (anderkonzen) [inline pt/lessons/basics/strings.md]: There are other "checar" and variants in the paragraphs above...

Other than that, good work! 👍

**Comment by ruan-brandao** [comment]
@anderkonzen I've updated the translation with your suggestions.

File: pt/lessons/basics/strings.md [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+  def sort_string(string) do
+    string
+    |> String.downcase
+    |> String.graphemes
+    |> Enum.sort
+  end
+end
+```
+
+Primeiro vamos olhar para `anagrams?/2`. Estamos checando se os parâmetros que recebemos são binários ou não. Essa é a forma de verificar se uma parâmetro é uma String em Elixir.
```
Comment (anderkonzen) [inline pt/lessons/basics/strings.md]: I found one more "checando" here!

**Comment by ruan-brandao** [comment]
@anderkonzen and @doomspork are we good to go with this one?

**Comment by doomspork** [comment]
@ruan-brandao sorry for the delays, I need to defer to @anderkonzen or @pragmaticivan on these translations.  If we don't hear them within the next or so we'll merge 👍

**Comment by anderkonzen** [comment]
@doomspork LGTM! :shipit:

**Comment by doomspork** [comment]
Thank you @anderkonzen 👍

**Pull request merged**